features/machine/machine.feature

```diff
@@ -225,3 +225,43 @@ Feature: Machine features testing
       | name     | <%= cb.noderef_name %> |
     Then the step should succeed
     And the output should match "node-role.kubernetes.io/infra="
+
+  # @author miyadav@redhat.com
+  # @case_id OCP-32269
+  @admin
+  @destructive
+  Scenario Outline: Implement defaulting machineset values for AWS/GCP/Azure/Vsphere
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+    And I use the "openshift-machine-api" project
+    Then admin ensures machine number is restored after scenario
+
+    #Below is needed for azure
+    Given I store the last provisioned machine in the :machine clipboard
+    Then evaluation of `machine(cb.machine).azure_location` is stored in the :default_location clipboard
+    #Below is needed for GCP
+    And evaluation of `machine(cb.machine).gcp_region` is stored in the :default_region clipboard
+    And evaluation of `machine(cb.machine).gcp_zone` is stored in the :default_zone clipboard
+    #Below is needed for AWS
+    And evaluation of `machine(cb.machine).aws_ami` is stored in the :default_ami clipboard
+    And evaluation of `machine(cb.machine).aws_availabilityzone` is stored in the :default_availabilityzone clipboard
+
+    Given I obtain test data file "cloud/<path>/ms_default_values.yaml"
+    When I run oc create over "ms_default_values.yaml" replacing paths:
+      | n                                                                                         | openshift-machine-api                           |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]           | <%= infrastructure("cluster").infra_name %>     |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"]        | default-valued-<path>                           |
+      | ["spec"]["template"]["metadata"]["labels"]["machine.openshift.io/cluster-api-cluster"]    | <%= infrastructure("cluster").infra_name %>     |
+      |  <extra_specs_1>                                                                          | <value_1>                                       |
+      |  <extra_specs_2>                                                                          | <value_2>                                       |
+      | ["spec"]["template"]["metadata"]["labels"]["machine.openshift.io/cluster-api-machineset"] | default-valued-<path>                           |
+    Then the step should succeed
+    And admin ensures "default-valued-<path>" machineset is deleted after scenario
```

Comment:
clean-up step should be registered just before creating the object. To make sure that even if step partially fails, we will remove machine set